data/src/test/java/org/apache/iceberg/io/TestTaskEqualityDeltaWriter.java

```diff
@@ -479,6 +479,11 @@ private GenericEqualityDeltaWriter(PartitionKey partition, Schema schema, Schema
       protected StructLike asStructLike(Record row) {
         return row;
       }
+
+      @Override
+      protected Record asStructLikeKey(Record data) {
+        return data;  // Note - this is correct as the whole schema is part of the key.
```

Comment:
Why not just say we don't support the `asStructLikeKey` in this testing writer ( which is better than adding a incorrect implementation ) ?

Reply:
That sounds reasonable to me. Throw `UnsupportedOperationException`?

Reply:
Yeah I'm fine with that. Sorry I missed this comment before.

Also, I'm not sure why tests are failing as they're passing for me. Maybe the cluster needs to be created with more slots for this many concurrent jobs? I'll see.

Reply:
If need be, I think we can do the same projection within this class. But updated to `UnsupportedOperationException`. I agree it's a bad idea to include code that just happens to work as people go to tests for context.

Reply:
As this is needed, I have updated it to create a new record off of the equalityDeleteSchema and then project onto those fields.

Let me know if there's a different way you think it should be done, but this should work even if the equality fields were only a subset of the fields.